pkg/virt-controller/watch/vm.go

```diff
@@ -546,23 +555,30 @@ func (c *VMController) handleVolumeRequests(vm *virtv1.VirtualMachine, vmi *virt
 	for i, request := range vm.Status.VolumeRequests {
 		vmCopy.Spec.Template.Spec = *controller.ApplyVolumeRequestOnVMISpec(&vmCopy.Spec.Template.Spec, &vm.Status.VolumeRequests[i])
 
-		if vmi != nil && vmi.DeletionTimestamp == nil {
-			if request.AddVolumeOptions != nil {
-				_, exists := vmiVolumeMap[request.AddVolumeOptions.Name]
-				if !exists {
-					err := c.clientset.VirtualMachineInstance(vmi.Namespace).AddVolume(vmi.Name, request.AddVolumeOptions)
-					if err != nil {
-						return err
-					}
-				}
-			} else if request.RemoveVolumeOptions != nil {
-				_, exists := vmiVolumeMap[request.RemoveVolumeOptions.Name]
-				if exists {
-					err := c.clientset.VirtualMachineInstance(vmi.Namespace).RemoveVolume(vmi.Name, request.RemoveVolumeOptions)
-					if err != nil {
-						return err
-					}
-				}
+		if vmi == nil || vmi.DeletionTimestamp != nil {
+			continue
+		}
+
+		if request.AddVolumeOptions != nil {
+			_, exists := vmiVolumeMap[request.AddVolumeOptions.Name]
+			if exists {
+				continue
+			}
```

Comment:
This can be inlined in the if condition:
`if _, exisits := vmiVolumeMap[request.AddVolumeOptions.Name]; exists {...}`